app/lib/summary_stats_utils.rb

```diff
@@ -0,0 +1,76 @@
+##
+# PortalUtils: generic class with server stats/maintenance methods
+#
+# note: all instances of :start_date and :cutoff_date are inclusive
+##
+class SummaryStatsUtils
+
+  # get a snapshot of user counts/activity up to a given date
+  # will give count of users as of that date, and number of active users on that date
+  def self.total_and_active_user_counts(cutoff_date: Date.today)
+    # make sure to make cutoff_date one day forward to include any users that were created on cutoff date
+    next_day = cutoff_date + 1.day
+    total_users = User.where(:created_at.lte => next_day).count
+    active_users = User.where(:current_sign_in_at => (cutoff_date..next_day)).count
+    {total: total_users, active: active_users}
+  end
+
+  # get a count of all submissions launch from the portal in a given 2 week period
+  # defaults to a time period of the last two weeks from right now
+  def self.analysis_submission_count(start_date: DateTime.now - 2.weeks, cutoff_date: DateTime.now)
+    AnalysisSubmission.where(:submitted_on => (start_date..cutoff_date), submitted_from_portal: true).count
+  end
+
+  # get a count of all studies created on the requested day
+  def self.study_creation_count(cutoff_date: Date.today)
```

Comment:
either rename to daily_study_creation_count or make it take a range like the analysis_submission_count method